core/src/main/java/nu/yona/server/analysis/entities/ActivityCommentMessage.java

```diff
@@ -76,12 +85,20 @@ public static ActivityCommentMessage createThreadHeadInstance(UUID senderUserId,
 
 	public ActivityCommentMessage getSenderCopyMessage()
 	{
-		return this.senderCopyMessage;
+		return (this.senderCopyMessageList == null || this.senderCopyMessageList.isEmpty()) ? null
+				: this.senderCopyMessageList.get(0);
 	}
 
 	private void setSenderCopyMessage(ActivityCommentMessage senderCopyMessage)
 	{
-		this.senderCopyMessage = senderCopyMessage;
+		if (this.senderCopyMessageList == null)
```

Comment:
Suggestion: extract a util method `this.senderCopyMessageSingleton = SingletonUtil.setValue(this.senderCopyMessageList, senderCopyMessage)

Reply:
See previous comment.